### PR TITLE
WP-4421 strong mode 2.x

### PR DESCRIPTION
**Pull request by toddbeckman-wf**
# Problem

w_transport needs some lints.

# Solution

Add the lints and make analyzer happy

# Testing Suggestion

- [ ] CI passes

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Raven
Number of Findings: 0

File: lib/src/web_socket/common/w_socket.dart [hunk 1]
```diff
@@ -149,10 +149,10 @@ abstract class CommonWSocket extends Stream implements WSocket {
   @override
   StreamSubscription listen(void onData(event),
       {Function onError, void onDone(), bool cancelOnError}) {
+    // ignore: cancel_subscriptions
     var sub = _incoming.stream
         .listen(onData, onError: onError, cancelOnError: cancelOnError);
     _incomingSubscription = new WSocketSubscription(sub, onDone, onCancel: () {
-      sub.cancel();
```
Comment (evanweible-wf) [inline lib/src/web_socket/common/w_socket.dart]: So this always perplexed me, but I'm not able to recreate the expected behavior for streams if I cancel this proxied subscription here. The incoming stream _should_ be cleaned up, but this is probably worth some follow-up investigation at a later date.

**Comment by toddbeckman-wf** [comment]
+1 to Evan's change

**Comment by maxwellpeterson-wf** [comment]
+10
- CI passed

**Comment by maxwellpeterson-wf** [comment]
QA +1
- [x] Dev +1's
- [x] Dev/QA +10 with detail of what was tested
- [x] Tests created/updated 
- [x] All tests pass

@Workiva/release-management-pp

**Pull request merged**